src/hotspot/share/gc/g1/g1Trace.cpp

```diff
@@ -37,23 +37,23 @@
 class G1HeapRegionTypeConstant : public JfrSerializer {
 public:
   void serialize(JfrCheckpointWriter& writer) {
-  static const u4 nof_entries = G1HeapRegionTraceType::G1HeapRegionTypeEndSentinel;
-  writer.write_count(nof_entries);
-  for (u4 i = 0; i < nof_entries; ++i) {
-    writer.write_key(i);
-    writer.write(G1HeapRegionTraceType::to_string((G1HeapRegionTraceType::Type)i));
+    static const u4 nof_entries = G1HeapRegionTraceType::G1HeapRegionTypeEndSentinel;
+    writer.write_count(nof_entries);
+    for (u4 i = 0; i < nof_entries; ++i) {
+      writer.write_key(i);
+      writer.write(G1HeapRegionTraceType::to_string((G1HeapRegionTraceType::Type)i));
+    }
   }
-}
 };
 
 class G1YCTypeConstant : public JfrSerializer {
 public:
   void serialize(JfrCheckpointWriter& writer) {
-    static const u4 nof_entries = G1GCYoungPhaseEndSentinel;
+    static const u4 nof_entries = static_cast<u4>(EnumRange<G1GCPauseType>{}.size());
```

Comment:
To avoid creating the `EnumRange`twice you could do:
```
constexpr EnumRange<G1GCPauseType> types{};
static const u4 nof_entries = static_cast<u4>(types.size());
```
And then use `types` in the for-loop below as well.